carta/image.py

```diff
@@ -672,6 +672,224 @@ def hide_contours(self):
+        if debiasing is None:
+            debiasing = self.macro("vectorOverlayConfig", "debiasing")
+        if q_error is None:
+            q_error = self.macro("vectorOverlayConfig", "qError")
+        if u_error is None:
+            u_error = self.macro("vectorOverlayConfig", "uError")
+        self.call_action("vectorOverlayConfig.setVectorOverlayConfiguration", angular_source, intensity_source, pixel_averaging_enabled, pixel_averaging, fractional_intensity, threshold_enabled, threshold, debiasing, q_error, u_error)
+
+    @validate(NoneOr(Number()), NoneOr(Number(), Constant(Auto)), NoneOr(Number(), Constant(Auto)), NoneOr(Number()), NoneOr(Number()), NoneOr(Number()))
+    def set_vector_overlay_style(self, thickness=None, intensity_min=None, intensity_max=None, length_min=None, length_max=None, rotation_offset=None):
+        """Set the styling (line thickness, intensity range, line length range, rotation offset) of vector overlay.
+
+        Parameters
+        ----------
+        thickness : {0}
+            The line thickness in pixels. By default is 1.
+        intensity_min : {1}
+            The minimum value of intensity in Jy/pixel. Use :obj:`carta.constants.Auto.AUTO` to clear the custom value and calculate it automatically.
+        intensity_max : {2}
+            The maximum value of intensity in Jy/pixel. Use :obj:`carta.constants.Auto.AUTO` to clear the custom value and calculate it automatically.
+        length_min : {3}
+            The minimum value of line length in pixels. By default is 0.
+        length_max : {4}
+            The maximum value of line length in pixels. By default is 20.
+        rotation_offset : {5}
+            The rotation offset in degrees. By default is 0.
+        """
+        if thickness is not None:
+            self.call_action("vectorOverlayConfig.setThickness", thickness)
+        if intensity_min is not None and intensity_max is not None:
+            self.call_action("vectorOverlayConfig.setIntensityRange", intensity_min, intensity_max)
+        elif intensity_min is None and intensity_max is not None:
+            self.call_action("vectorOverlayConfig.setIntensityRange", self.macro("vectorOverlayConfig", "intensityMin"), intensity_max)
+        elif intensity_min is not None and intensity_max is None:
+            self.call_action("vectorOverlayConfig.setIntensityRange", intensity_min, self.macro("vectorOverlayConfig", "intensityMax"))
+        elif intensity_min is Auto.AUTO and intensity_max is Auto.AUTO:
+            self.call_action("vectorOverlayConfig.setIntensityRange", Undefined(), Undefined())
+        elif intensity_min is not Auto.AUTO and intensity_max is Auto.AUTO:
+            self.call_action("vectorOverlayConfig.setIntensityRange", intensity_min, Undefined())
+        elif intensity_min is Auto.AUTO and intensity_max is not Auto.AUTO:
+            self.call_action("vectorOverlayConfig.setIntensityRange", Undefined(), intensity_max)
```

Comment:
It's unnecessary to handle all these specific combinations (and you're currently not handling possible combinations of auto and None). You can simplify this logic a lot:

1. check if `intensityMin` is `None`, and if so set it to the macro to fetch the current value
2. otherwise check if it's `Auto.AUTO`, and if so set it to the undefined macro
3. do the same for `intensityMax`
4. then call the action with both (updated) values.